src/store/plugins/router.js

```diff
@@ -50,15 +50,32 @@ export default store => {
   })
 
   router.beforeEach(async (to, from, next) => {
+    for (let m of from.matched) {
+      if (m.meta.onLeave) {
+        console.log('onLeave', m.meta.onLeave)
+        store.dispatch(m.meta.onLeave)
+      }
+    }
+    for (let m of to.matched) {
+      if (m.meta.beforeEnter) {
+        try {
+          console.log('beforeEnter', to.params.groupInfoId)
+          // TODO parse all params, assuming that all are integer
+          const params = { groupInfoId: parseInt(to.params.groupInfoId, 10) }
+          store.dispatch(m.meta.beforeEnter, params)
+        }
+        catch (error) {
+          console.log('onError', m.meta.onError)
+          store.dispatch('routeError/set', { translation: m.meta.onError })
```

Comment:
Maybe find a way here to catch route errors specifically, and rethrow others. You can make custom error classes :) Or just add certain data to a normal error. Then get the translation key from the thrown error.